### DISCUSSION TITLE
Could not establish connection to the Collabora Online server

### DISCUSSION THREAD

**Discussion by vawaver** [nextcloud/all-in-one#1312] [Questions] [answered]
### Steps to reproduce
1. myDomain ->my publicIP ->myRouter->myProxmox->myVirtualMachine (Ubuntu server 22.04) - 10 GB RAM
All required ports are forwarded from my router directly to VirtualMachine's local IP address.
2. I followed this tutorial - https://www.youtube.com/watch?v=Nh2-LjIymmQ
3. -e SKIP_DOMAIN_VALIDATION=true  - used in docker command
4. I installed AIO but it is impossible to enable Nextcloud Office because Error: Could not establish connection to the Collabora Online server for the same domain as AIO installation.

#### Other valuable info  
![image](https://user-images.githubusercontent.com/52236863/197825726-f93a5e57-a10a-4e36-9380-ebdf140d791d.png)
![image](https://user-images.githubusercontent.com/52236863/197826415-2219cbad-fbe5-4449-8c94-64cbe1e8bdd9.png)

**Comment by szaimen** [comment]
Moving to discussions since I cannot reproduce this

**Comment by szaimen** [comment]
Hi, i fear I cannot reproduce this.

Can you please post the nextcloud-aio-nextcloud container logs here? 

Also are you running AIO behind a reverse proxy and/or are you using docker rootless or podman?

**Comment by vawaver** [comment]
Hi,
I attached the log.
[aio_log.txt](https://github.com/nextcloud/all-in-one/files/9862799/aio_log.txt)

I am not behind reverse Proxy now. .... but I will use NGINX Proxy later (I found your tutorial for that).

I started the docker command as sudo user, not root. I do not use podman.
Thank you.

**Comment by szaimen** [comment]
Thanks for the response! However I need the logs of the `nextcloud-aio-nextcloud` container and not `nextcloud-aio-mastercontainer`!

**Comment by vawaver** [comment]
Sorry, my bad.
I hope it is what you asked for.
[_nextcloud-aio-nextcloud_logs.txt](https://github.com/nextcloud/all-in-one/files/9862962/_nextcloud-aio-nextcloud_logs.txt)

Thank you.

**Comment by szaimen** [comment]
Thanks! Can you click on that button and check if that makes it work?
![image](https://user-images.githubusercontent.com/42591237/197854308-febf17fa-f213-42f3-969a-0ff1fb82a025.png)

**Reply by vawaver** [reply]
Of course I tried to use "Save" button before. And because it did not help I created my issue.
Or do you mean I try to Save now and after that I will send you a log again?

**Reply by szaimen** [reply]
So the save button does not work? Can you click the button and post the logs of `https://yourdomain.com/settings/admin/logging` afterwards here?

**Comment by vawaver** [comment]
I hope I am sending the right one.
[save_log.txt](https://github.com/nextcloud/all-in-one/files/9863165/save_log.txt)

**Comment by szaimen** [comment]
Hm... so it looks like websockets connections do not work in general. 

Which reminds me of this thread: https://github.com/nextcloud/all-in-one/discussions/1196.

I honestly do not think that we can improve AIO to fix this...

It is rather something with your domain or maybe web firewall and/or something in between AIO and the client...

Where are you hosting your domain and are you using some kind of web firewall or therelike?

Did you also already try different browsers?

---

EDIT: I googled the problem again and the issue may be an adblocker?

https://www.reddit.com/r/audiobookshelf/comments/ub021p/caddy_v2_as_reverse_proxy_websockets_not/

Sorry, I am honestly still tapping in the dark since the issue is not reproducible in my test instance. Everything works as expected here.

---

EDIT2: for google domains I found this: https://support.glitch.com/t/websockets-do-not-work-on-custom-domain/31171/5
For cloudflare I found this: https://support.cloudflare.com/hc/en-us/articles/200169466-Using-Cloudflare-with-WebSockets#12345680

**Comment by vawaver** [comment]
Hi,
thank you for your answer.
I try to explain the connection of the domain to my server. 

1. Local Domain seller sold me a domain - I had created a subdomain - for example nextcloud.mydomain.sk. I created A DNS record of the subdomain to my public IP (it is my static IP).
2. I do not use any external hosting provider. I have my own hardware server with Proxmox (firewall off).
3. Under Proxmox I created a virtual machine 1000GB space + 8 cores + 10GB RAM - Ubuntu server 22.04.1.
UFW firewall is disabled.
LAN IP for the server is 192.168.1.22
On the Ubuntu server I edited the hostname and hosts files as follows:
hosts
127.0.1.1 nextcloud.mydomain.sk

hostname
nextcloud.mydomain.sk

No other settings were applied in the Ubuntu server filesystem.

4. I have made port forwarding on my router as follows
ports 80,443,8443,3478 -> 192.168.1.22
Those are my settings.
5. Internet connection speed to my server - 1000Mbit/500Mbit - optical

I tried 2 browsers - Chromium a FIrefox (Ad Blockers - disabled) with the same results.
I did not succeed in setup the Collabora document server as I mentioned above.

I am not an IT pro... just an amateur so I do not understand all.

I do not use any Proxy service.
PS: Maybe I missed some required DNS settings on my Ubuntu server. or on Domain DNS settings.

Thank you for your answer.

**Reply by szaimen** [reply]
What happense when you remove that entry from your hosts file again? `127.0.1.1 nextcloud.mydomain.sk` but I dont think that this should make it work.

Feel free to try out the pointers that I've send. But as I said, I have no clue unfortunately as everything works on my test instance...

**Comment by vawaver** [comment]
I tried to remove the line above and It did not help.  

As you sent me the hints.
I am not using Adblocker in my web browser and really do not know what should else I try.
Perhaps I would like to know whether you have a working virtual image for Proxmox elsewhere that I can try to import.

**Comment by szaimen** [comment]
I just had an idea what you could try:

What is the output of
`docker exec -it nextcloud-aio-nextcloud curl -vvv https://yourdomain.com:443/standalone-signaling/api/v1/welcome`

**Comment by szaimen** [comment]
Also, what does `docker exec -it nextcloud-aio-nextcloud curl -vvv https://yourdomain.com:443/hosting/discovery` return?

**Comment by vawaver** [comment]
I tried to remove the line above and It did not help.  

As you sent me the hints.
I am not using Adblocker in my web browser and really do not know what should else I try.

I am sending output logs from both commands above.
[docker_command01.txt](https://github.com/nextcloud/all-in-one/files/9872002/docker_command01.txt)
[docker_command02.txt](https://github.com/nextcloud/all-in-one/files/9872003/docker_command02.txt)

**Reply by szaimen** [reply]
Hm... are the collabora and talk container running? The outputs do not seem to be complete or did you remove some of the output?

**Reply by szaimen** [reply]
Also, you obviously need to adjust the domain `yourdomain.com` to the domain that you are using for Nextcloud...

**Reply by vawaver** [reply]
> Hm... are the collabora and talk container running? The outputs do not seem to be complete or did you remove some of the output?

Outputs are completed. I did not remove anything.
![image](https://user-images.githubusercontent.com/52236863/198092139-bd34d323-dd2a-489e-b518-e69d7c6f4575.png)

**Reply by szaimen** [reply]
Interesting! Why is nextcloud-aio-apache unhealthy?

Can you post its logs here?

**Reply by szaimen** [reply]
Also please try the command again that I've send and adjust `yourdomain.com` to the domain that you are using for Nextcloud.

**Reply by vawaver** [reply]
> Also, you obviously need to adjust the domain `yourdomain.com` to the domain that you are using for Nextcloud...

Ufff, I did not get you. What should I exactly do?
Please be so kind and tell me step-by-step.
Thank you.

**Reply by vawaver** [reply]
> Interesting! Why is nextcloud-aio-apache unhealthy?
> 
> Can you post its logs here?

You mean log from Portainer -> Apache docker?

**Reply by vawaver** [reply]
I am sending apache logs.
[_nextcloud-aio-apache_logs.txt](https://github.com/nextcloud/all-in-one/files/9872202/_nextcloud-aio-apache_logs.txt)

**Reply by szaimen** [reply]
> Ufff, I did not get you. What should I exactly do?
> Please be so kind and tell me step-by-step.
> Thank you.

The two commands below contain `yourdomain.com` as domain. Please adjust the domain to the one that you initially entered in AIO that gets used by Nextcloud and post the output here.

> `docker exec -it nextcloud-aio-nextcloud curl -vvv https://yourdomain.com:443/standalone-signaling/api/v1/welcome`
> `docker exec -it nextcloud-aio-nextcloud curl -vvv https://yourdomain.com:443/hosting/discovery`

**Reply by vawaver** [reply]
> > Ufff, I did not get you. What should I exactly do?
> > Please be so kind and tell me step-by-step.
> > Thank you.
> 
> The two commands below contain `yourdomain.com` as domain. Please adjust the domain to the one that you initially entered in AIO that gets used by Nextcloud and post the output here.
> 
> > `docker exec -it nextcloud-aio-nextcloud curl -vvv https://yourdomain.com:443/standalone-signaling/api/v1/welcome`
> > `docker exec -it nextcloud-aio-nextcloud curl -vvv https://yourdomain.com:443/hosting/discovery`

Oh, I am dumb, sorry.
[docker_command02.txt](https://github.com/nextcloud/all-in-one/files/9872437/docker_command02.txt)
[docker_command01.txt](https://github.com/nextcloud/all-in-one/files/9872438/docker_command01.txt)

Sending both logs again.

**Comment by szaimen** [comment]
So it seems like you indeed have cloudflare in between the client and AIO based om the logs that you provided.

Can you try to disable the clouflare caching and check if that makes it work?

See e.g.https://community.cloudflare.com/t/disable-cloudflare-caching-completly/86417

**Comment by szaimen** [comment]
Can you please also try to get completely rid of cloudflare in the chain to eliminate that as a possible problem?

**Comment by szaimen** [comment]
I also found this: https://support.cloudflare.com/hc/en-us/articles/360050691831-Understanding-Network-Error-Logging

**Reply by vawaver** [reply]
Sorry for the late reply.
I think I found a solution to my problem. 
I am sending you a URL with today's YOUTUBE tutorial - https://youtu.be/OCLq62KOqNU
Here is exactly the docker-compose configuration that I was looking for.
AIO + NGINX
Can you please add the docker setting on your GitHub webpage? I think this helps all beginners with docker.

**Reply by Zoey2936** [reply]
> Sorry for the late reply. I think I found a solution to my problem. I am sending you a URL with today's YOUTUBE tutorial - https://youtu.be/OCLq62KOqNU Here is exactly the docker-compose configuration that I was looking for. AIO + NGINX Can you please add the docker setting on your GitHub webpage? I think this helps all beginners with docker.

Everything done in the Video is already documented (running AIO behind nginx-proxy-manager) - Also the creator of the video has set some useless environment variables. But it sounds good that running the AIO behind the npm fixed your problem.

**Comment by szaimen** [comment, marked as answer]
To sum this up: seems like switching to NPM solved the problem

**Reply by hakunamatata97k** [reply]
Hey @szaimen, according to you and the video a subdomain is used for reaching the Nextcloud instance as such `nextcloud.example.local` right? then how did your office domain look like? `office.example.local`? 
Somehow I managed to get a green check mark when entering the `nextcloud.example.local` in the Nextcloud office URL field, but this is counterintuitive! As the documentation states I have to get it to work with `office.example.local`! Could you please state what URL did you type in the Nextcloud office field? And did you have to extra configure an office subdomain in the Nginx proxy manager? :?

**Reply by szaimen** [reply]
> Somehow I managed to get a green check mark when entering the `nextcloud.example.local` in the Nextcloud office URL field

Actually the same domain is used also for Nextcloud Office. So this looks good.

**Comment by punkyard** [comment]
Hi,
moving to NPM didn't change a thing for me. Both Collabora and Talk won't work after install.
I have to add my public IP to Collabora WOPI request. No solution yet for Talk..
